### PR TITLE
Fix auth

### LINKED ISSUE #664: Bug with moderators
From Rizzrack on coinbistro:

Mods should be able to be assigned to  just one or two boards.

For example:

User X has moderator role ONLY for Economy - Mockup Market - subboard test. He should have mod permissions (delete post, edit post, move thread only from that sub-board somewhere else etc) and also see only the reports for posts from that sub-board.

Because he is assigned to that sub-board he should not be able to do those actions on it's parent boards or anywhere else for that matter.

Added user Coin_BoT to moderator role and assigned him only to Epochtalk for Bitcointalk.org and it's child board. However he can still delete posts in Meta and Economy boards. I did not see any option to limit this to one board/sub-board in particular.

Mod actions that should need to be limited ONLY to selected boards:
delete/edit/lock user post
delete/lock thread
move thread (however if it is moved outside his board he will no longer have access to move it again)
sticky a thread
view/handle post reports

What does not work properly on Epochtalk:
in admin panel > moderation > posts there is a "show post reports for all boards" button. Need to remove it. Mods on specific boards need to handle reports only for those boards. Or maybe make it view only. But needs slight tweaking... Relevant commit: https://github.com/epochtalk/epochtalk/commit/fc19d7d11442b6614e7e79af1cf2d3e150b33f28
you can edit posts in other boards than the ones assigned to you

### LINKED ISSUE #685: priority for global mods over board mods
i get a 403 error when using a global moderator account to update the post content of a board moderator

the error message is `owner`

(resolved in #671)

### PR DESCRIPTION
**Pull request by akinsey**
Test Procedure:
1) Test that moderators can only moderate users in their board, with a lower priority than moderator
2) Test that global moderators can moderate users in any board, as long as they are lower priority than a global moderator
3) Test that all moderators types only see icons for actions they can perform
4) Ensure self mod still works as expected

Fixes #664 
Fixes #685

### CODE REVIEW

**Comment by unenglishable** [comment]
checked nested boards and permissions; here are my findings:

* board moderation does not cascade down to child boards

* if a user has the moderator

**Comment by akinsey** [comment]
New Fixes:
- Moderators actions should now be restricted based on priority, both post and thread actions (except move thread)
- Moderators should now be able to moderate moderators of other boards
- Global moderator actions should now be restricted based on priority, both post and thread actions (except move thread)
- These actions should be restricted in the UI and on the Route Auth

NOTES: Moderation does not cascade down to child boards.

New Test Procedure:
1) Reset the moderator and global moderator roles in the admin panel. This will load new permissions
2) Create posts and threads as both an admin and a regular user
3) Attempt to moderate those posts/threads with both a moderator and global moderator account
4) Moderators now need to have priority over a user to moderate them
5) Ensure self-moderated threads still work

@unenglishable @boka

**Comment by unenglishable** [comment]
@akinsey i get a 403 error when using a `global moderator` account to update the post content of a board `moderator`

the error message is `owner`

File: modules/ept-posts/authorization/delete.js [hunk 1]
```diff
@@ -60,6 +60,25 @@ module.exports = function postsDelete(server, auth, postId) {
...
+      permission: 'posts.delete.bypass.owner.mod'
+    },
+    {
+      type: 'runValidation',
+      method: common.hasPriority,
+      args: [server, auth, 'posts.delete.bypass.owner.mod', postId]
+    }
+  ];
+  var standardMod = server.authorization.stitch(Boom.forbidden(), standardModCond, 'all');
+
```
Comment (unenglishable) [inline modules/ept-posts/authorization/delete.js]: extra line here

File: modules/ept-posts/client/posts/posts.controller.js [hunk 2]
```diff
@@ -16,6 +16,10 @@ var ctrl = [
     this.user = Session.user;
     this.posts = pageData.posts;
     this.thread = pageData.thread;
+    this.moderators = pageData.board.moderators.map(function(data) {
+      return data.id;
+    });
+
```
Comment (unenglishable) [inline modules/ept-posts/client/posts/posts.controller.js]: get rid of this line too maybe?

**Pull request merged**